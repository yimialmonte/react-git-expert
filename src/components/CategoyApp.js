import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoyApp = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 1) {
      setCategories((prev) => [...prev, inputValue]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInput} />
    </form>
  );
};

CategoyApp.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default CategoyApp;
