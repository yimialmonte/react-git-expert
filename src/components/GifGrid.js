import React, { useEffect, useState } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGif';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then(setImages);
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((item) => (
          <GifGridItem key={item.id} {...item} />
        ))}
      </div>
    </>
  );
};
