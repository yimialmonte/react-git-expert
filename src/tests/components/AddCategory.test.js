import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategoy'

describe('Test on AddCategory Component', () => {

  const setCategories = () => {};

  test('Should show component correctly', () => {
    const wrapper = shallow( <AddCategory setCategories={setCategories} />);
    expect(wrapper).toMatchSnapshot();
  })

});
