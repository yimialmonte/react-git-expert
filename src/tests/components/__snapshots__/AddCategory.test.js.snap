// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on AddCategory Component Should show component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
